soc/nordic/nrf53/soc.c

```diff
@@ -490,7 +517,9 @@ static int nordicsemi_nrf53_init(void)
 	nrf_gpio_pin_control_select(PIN_XL2, NRF_GPIO_PIN_SEL_PERIPHERAL);
 #endif /* !defined(CONFIG_BUILD_WITH_TFM) */
 #endif /* defined(CONFIG_SOC_ENABLE_LFXO) */
-#if defined(CONFIG_SOC_HFXO_CAP_INTERNAL)
+#if defined(CONFIG_SOC_HFXO_CAP_INTERNAL) || \
+	(defined(NRF_SOC_SECURE_SUPPORTED) && \
+	 DT_ENUM_HAS_VALUE(DT_NODELABEL(hfxo), load_capacitors, internal))
```

Comment:
Also here `#if defined(HFXO_CAP_VAL_X2)` could be used (`HFXO_CAP_VAL_X2` could be defined only when `CONFIG_NRF_SOC_SECURE_SUPPORTED`).

Reply:
thanks for the optimization, updated